spec/unit/type/user_spec.rb

```diff
@@ -325,24 +325,31 @@ def self.instances; []; end
   end
 
   describe "when managing passwords" do
-    before do
-      @password = described_class.new(:name => 'foo', :password => 'mypass').parameter(:password)
-    end
+    let(:transaction) { Puppet::Transaction.new(Puppet::Resource::Catalog.new, nil, nil) }
+    let(:harness) { Puppet::Transaction::ResourceHarness.new(transaction) }
+    let(:resource) { described_class.new(:name => 'foo', :password => 'mypass') }
 
     it "should not include the password in the change log when adding the password" do
-      expect(@password.change_to_s(:absent, "mypass")).not_to be_include("mypass")
+      status = harness.evaluate(resource)
+      sync_event = status.events[0]
+      puts status.events
+      expect(sync_event.message).to eq 'changed [redacted] to [redacted]'
     end
 
     it "should not include the password in the change log when changing the password" do
-      expect(@password.change_to_s("other", "mypass")).not_to be_include("mypass")
+      harness.evaluate(resource)
+      resource[:password] = 'super secret'
+      status = harness.evaluate(resource)
+      sync_event = status.events[0]
+      expect(sync_event.message).to eq 'changed [redacted] to [redacted]'
     end
 
     it "should redact the password when displaying the old value" do
-      expect(@password.is_to_s("currentpassword")).to match(/^\[old password hash redacted\]$/)
+      expect(resource.parameter(:password).is_to_s("currentpassword")).to match(/^\[old password hash redacted\]$/)
     end
 
     it "should redact the password when displaying the new value" do
-      expect(@password.should_to_s("newpassword")).to match(/^\[new password hash redacted\]$/)
+      expect(resource.parameter(:password).should_to_s("newpassword")).to match(/^\[new password hash redacted\]$/)
```

Comment:
@jhelwig these tests are now broken with our change since it's testing the `should_to_s` method, but I can't visualize the correct thing to change them too. Do you have any guidance of where something else is already testing what we should be testing here?